src/blockchain/database/common/BlockFilter.cpp

```diff
@@ -131,25 +127,21 @@ auto BlockFilter::LoadFilters(
         (1000u * sizeof(util::IndexData)) + sizeof(Vector<util::IndexData>);
     auto buf = std::array<std::byte, allocBytes>{};
     auto alloc = alloc::BoostMonotonic{buf.data(), buf.size()};
-    const auto indices = [&] {
-        auto out = Vector<util::IndexData>{&alloc};
-        out.reserve(1000u);
-        auto tx = lmdb_.TransactionRO();
-
-        for (const auto& hash : blocks) {
-            try {
-                auto& index = out.emplace_back();
-                load_filter_index(type, hash.Bytes(), tx, index);
-            } catch (const std::exception& e) {
-                LogVerbose()(OT_PRETTY_CLASS())(e.what()).Flush();
-                out.pop_back();
-
-                break;
-            }
-        }
 
-        return out;
-    }();
+    Vector<util::IndexData> indices{&alloc};
+    indices.reserve(1000u);
+    auto tx = lmdb_.TransactionRO();
```

Comment:
I suggest a pair of braces from here to line 144 inclusive, to force the destruction of tx as soon as possible - it is a locked transaction.